lib/src/replay/screenshot/screenshot_capturer.dart

```diff
@@ -86,79 +86,135 @@ class ScreenshotCapturer {
 
     final viewId = identityHashCode(renderObject);
 
+    final Completer<ImageInfo?> completer = Completer<ImageInfo?>();
+
     try {
       final srcWidth = renderObject.size.width;
       final srcHeight = renderObject.size.height;
       final pixelRatio =
           _getPixelRatio(srcWidth: srcWidth, srcHeight: srcHeight);
 
-      final ui.Image image = await renderObject.toImage(pixelRatio: pixelRatio);
+      final syncImage = renderObject.toImage(pixelRatio: pixelRatio);
 
       final replayConfig = _config.sessionReplayConfig;
 
-      // using png because its compressed, the native SDKs will decompress it
-      // and transform to jpeg if needed (soon webp)
-      // https://github.com/brendan-duncan/image does not have webp encoding
-      Uint8List? pngBytes = await _getImageBytes(image);
-      if (pngBytes == null || pngBytes.isEmpty) {
-        printIfDebug('Error: Failed to convert image byte data to Uint8List.');
-        image.dispose();
-        return null;
+      // call getCurrentScreenRects if really necessary
+      List<ElementData>? elementsDataWidgets;
+      if (replayConfig.maskAllTexts || replayConfig.maskAllImages) {
+        elementsDataWidgets =
+            PostHogMaskController.instance.getCurrentWidgetsElements();
       }
 
-      if (const PHListEquality().equals(pngBytes, statusView.imageBytes)) {
-        printIfDebug(
-            'Debug: Snapshot is the same as the last one, nothing changed, do nothing.');
-        image.dispose();
-        return null;
-      }
+      /// we firstly get current image (syncImage) and masks
+      /// (postHogWidgetWrapperElements, elementsDataWidgets) synchronously and
+      /// then executed the main process asynchronous
+      Future(() async {
+        final isSessionReplayActive =
+            await _nativeCommunicator.isSessionReplayActive();
+        if (!isSessionReplayActive) {
+          return;
+        }
 
-      statusView.imageBytes = pngBytes;
+        final recorder = ui.PictureRecorder();
+        final canvas = Canvas(recorder);
+        final image = await syncImage;
```

Comment:
```suggestion
        final image = await renderObject.toImage(pixelRatio: pixelRatio);
```

would that work?

Reply:
with 

```dart
final image = await renderObject.toImage(pixelRatio: pixelRatio);
```

the renderObject reference may have changed in the main thread, so it would be out of sync

Reply:
can we do the whole process inside of the Future then? so its all done in sequence

Reply:
But if you have a solution to the whole process in the Future block, I would like to see it, maybe I missed something or something is beyond my ability haha

Reply:
everything right now is being done on the main thread, Dart is single-threaded, unless you use [background workers](https://dart.dev/language/concurrency#background-workers) which is not available in all Dart/Flutter versions.
Maybe you are misunderstanding the concept of async/await with multithreading?

Reply:
@thisames something like this [`9992590` (#150)](https://github.com/PostHog/posthog-flutter/pull/150/commits/99925900e4d6d8eb918473b8dba6484ca96eff81)
can you check if this works?

Reply:
sure, I tested with this code, see the result

https://github.com/user-attachments/assets/7f6aae7a-7dbd-4df3-afdb-e2746635d647

Reply:
about isolate, I think it wouldn't work because `RenderRepaintBoundary` `toImage()` makes calls that depend on the Flutter engine to draw on a GPU surface (or internal buffer). This process all occurs within the rendering engine, which is already linked to the main isolate. I believe that calling a secondary isolate for this wouldn't work...

Reply:
yep, i was just explaining the usage of background workers and isolates, but this has to be done on the main isolate.